### PR TITLE
Laboratory work #3, Polina Zhilina - 21FPL2 

### PR DESCRIPTION
**Pull request by polyamba**

### CODE REVIEW

File: lab_2_keywords_cooccurrence/main.py [hunk 1]
```diff
@@ -270,3 +270,4 @@ def process_text(text: str, stop_words: Optional[Sequence[str]] = None, max_leng
     if keyword_phrases_with_scores and cumulative_score_with_stop_words is not None:
         return {**keyword_phrases_with_scores, **cumulative_score_with_stop_words}
     return None
+
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:273]: revert

File: lab_2_keywords_cooccurrence/start.py [hunk 2]
```diff
@@ -70,4 +70,4 @@ def read_target_text(file_path: Path) -> str:
 
     RESULT = UNKNOWN_PROCESSED
 
-    assert RESULT, 'Keywords are not extracted'
+    assert RESULT, 'Keywords are not extracted'
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/start.py:73]: revert

File: lab_2_keywords_cooccurrence/target_score.txt [hunk 3]
```diff
@@ -1 +1 @@
-10
\ No newline at end of file
+10
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/target_score.txt:1]: revert

File: lab_3_keywords_textrank/README.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # Лабораторная работа №3
-  
+
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/README.md:2]: revert

File: lab_3_keywords_textrank/main.py [hunk 5]
```diff
@@ -51,7 +52,13 @@ def _clean_and_tokenize(self, text: str) -> tuple[str, ...]:
             tuple[str, ...]
                 clean lowercase tokens
         """
-        pass
+        clean_text = ''
+        for el in text:
+            if el not in self._punctuation:
+                clean_text += el
+        clean_text = clean_text.lower()
+        tokens = tuple(clean_text.split())
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:60]: return the tuple without creating a variable

File: lab_3_keywords_textrank/main.py [hunk 6]
```diff
@@ -81,7 +92,9 @@ def preprocess_text(self, text: str) -> tuple[str, ...]:
             tuple[str, ...]
                 clean lowercase tokens with no stop-words
         """
-        pass
+        tokens = self._clean_and_tokenize(text)
+        no_stopwords = self._remove_stop_words(tokens)
+        return no_stopwords
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:97]: again. Do not create unnecessary variable

File: lab_3_keywords_textrank/main.py [hunk 7]
```diff
@@ -120,7 +134,13 @@ def _learn_indices(self, tokens: tuple[str, ...]) -> None:
             tokens : tuple[str, ...]
                 sequence of string tokens
         """
-        pass
+        i = 0
+        for token in tokens:
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:138]: strange implementation. Use enumerate

File: lab_3_keywords_textrank/main.py [hunk 8]
```diff
@@ -136,7 +156,14 @@ def encode(self, tokens: tuple[str, ...]) -> Optional[tuple[int, ...]]:
                 sequence of integer tokens
         In case of empty tokens input data, None is returned
         """
-        pass
+        if not tokens:
+            return None
+        self._learn_indices(tokens)
+        tuple_numb = []
+        for token in tokens:
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:163]: for + cast to tuple - in one line. Use tuple comprehension

File: lab_3_keywords_textrank/main.py [hunk 9]
```diff
@@ -233,7 +276,21 @@ def add_edge(self, vertex1: int, vertex2: int) -> int:
...
         """
-        pass
+        if vertex1 == vertex2:
+            return -1
+
+        for vertex in vertex1, vertex2:
+            if vertex not in self._list_of_vertexes:
+                self._list_of_vertexes.append(vertex)
+                self._matrix.append([])
+        for i in self._matrix:
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:286]: how to make it faster

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Review by demid5111 — APPROVED** [review]

**Pull request closed without merging**